apps/studio/src/background/theme-handlers.ts

```diff
@@ -0,0 +1,267 @@
+      console.error('Error getting theme:', error);
+      return { success: false, error: error.message };
+    }
+  });
+
+  // Apply a custom theme
+  ipcMain.handle('theme/apply', async (event, { themeId }) => {
+    try {
+      const theme = customThemes[themeId];
+
+      if (!theme) {
+        return { success: false, error: 'Theme not found' };
+      }
+
+      const webContents = event.sender;
+      const winId = webContents.id;
+
+      // Apply the CSS to the renderer
+      await applyThemeCSS(webContents, themeId, theme.css);
+
+      // Remember which theme is applied to this window
+      windowThemes[winId] = themeId;
+
+      return { success: true };
+    } catch (error) {
+      console.error('Error applying theme:', error);
+      return { success: false, error: error.message };
+    }
+  });
+
+  // List all custom themes
+  ipcMain.handle('theme/list', async () => {
+    try {
+      const themes = Object.values(customThemes).map(theme => ({
+        id: theme.id,
+        name: theme.name,
+        type: theme.type,
+        colors: theme.colors
+      }));
+
+      return { success: true, themes };
+    } catch (error) {
+      console.error('Error listing themes:', error);
+      return { success: false, error: error.message };
+    }
+  });
+
+  // Remove a custom theme
+  ipcMain.handle('theme/remove', async (event, { themeId }) => {
+    try {
+      if (!customThemes[themeId]) {
+        return { success: false, error: 'Theme not found' };
+      }
+
+      // Remove the theme from all windows that use it
+      const webContents = event.sender;
+      const winId = webContents.id;
+
+      if (windowThemes[winId] === themeId) {
+        await removeThemeCSS(webContents, themeId);
+        delete windowThemes[winId];
+      }
+
+      // Remove the theme from storage
+      delete customThemes[themeId];
+
+      return { success: true };
+    } catch (error) {
+      console.error('Error removing theme:', error);
+      return { success: false, error: error.message };
+    }
+  });
+}
+
+// Apply theme CSS to a window
+async function applyThemeCSS(
+  webContents: WebContents,
+  themeName: string,
+  css: string
+) {
+  try {
+    console.log(`Applying CSS for theme ${themeName} to window ${webContents.id}`);
+
+    // First remove any existing theme
+    if (windowThemes[webContents.id]) {
+      await removeThemeCSS(webContents, windowThemes[webContents.id]);
+    }
+
+    // Make sure we have valid CSS
+    if (!css || css.trim() === '') {
+      console.error(`Empty CSS for theme ${themeName}`);
+      return null;
+    }
+
+    // Inject the CSS
+    const key = await webContents.insertCSS(css, { cssOrigin: 'user' });
+    console.log(`CSS injected for theme ${themeName} with key ${key.substring(0, 10)}...`);
+
+    // Store the key for later removal
+    themeKeys[`${webContents.id}_${themeName}`] = key;
+
+    // Also set the theme class directly via JavaScript
+    await webContents.executeJavaScript(`
```

Comment:
This doesn't seem like something we should be doing in a handler. Can the CSS and theme name not be passed back to the front end, and we have the front-end apply it?

Reply:
totally